hail/python/test/hail/table/test_table.py

```diff
@@ -837,6 +837,35 @@ def test_aggregate_localize_false(self):
         ht = ht.annotate(y = ht.idx + ht.aggregate(hl.agg.max(ht.idx), _localize=False))
         assert ht.y.collect() == [x + 9 for x in range(10)]
 
+    def test_same_different_type(self):
```

Comment:
There should probably be a positive test of `_same`, to be sure it doesn't just always return `false`, maybe with a case with tables that are not identical but within the tolerance.